### PR TITLE
Remove submodules from client

### PR DESCRIPTION
**Pull request by dannygoldstein**
Description
-----------

Removes `sweeps_engine` submodule and replaces it with a vendored release of `github.com/wandb/sweeps`. The vendoring is handled automatically by a script and verified in CI by circle. 

Testing
-------

How was this PR tested?

Release Notes
-------------

Below, please enter user-facing release notes as one or more bullet points.
If your change is not user-visible, write `NO RELEASE NOTES` instead, with no bullet points.

------------- BEGIN RELEASE NOTES ------------------
NO RELEASE NOTES
------------- END RELEASE NOTES --------------------

### CODE REVIEW

**Review by vanpelt — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ workflows:
       - test:
          name: "code-check"
          image: "python:3.6"
-         toxenv: "protocheck,mypy,black,flake8,flake8-docstrings"
+         toxenv: "protocheck,mypy,black,flake8,flake8-docstrings,verify-sweeps"
```
Comment (dannygoldstein) [inline .circleci/config.yml:81]: This tox target makes sure that the sweeps version that was vendored to `wandb/sweeps` matches the contents of the https://github.com/wandb/sweeps git repository @ src/sweeps and that github.com/wandb/sweeps requirements.txt matches requirements.sweeps.txt in this repo

File: .sweeps_ref [hunk 2]
```diff
@@ -0,0 +1 @@
+v0.0.5
```
Comment (dannygoldstein) [inline .sweeps_ref:1]: This is the version of the sweeps repo that we have pinned to

File: Makefile [hunk 3]
```diff
@@ -94,3 +85,5 @@ bumpversion-to-dev:
 bumpversion-from-dev:
 	tox -e bumpversion-from-dev
 
+vendor-sweeps:  # expects an argument called ref=,  example usage: make vendor-sweeps ref=master
+	bash tools/vendor_sweeps.sh $(ref)
```
Comment (dannygoldstein) [inline Makefile:89]: This makefile target clones the sweeps repository to a temporary directory, checks out the ref pointed to by $ref, then copies over the requirements.txt file (to install the sweeps extras in wandb/client) and the library contents to wandb/sweeps

File: requirements.sweeps.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+numpy>=1.15,<1.21
+scipy>=1.5.4
+PyYAML
+scikit_learn==0.24.1
+jsonschema>=3.2.0
+jsonref>=0.2
+pydantic>=1.8.2
```
Comment (dannygoldstein) [inline requirements.sweeps.txt:7]: the contents of this file is vendored using `bash tools/vendor_sweeps.sh`. It is automatically verified in circle using `tox -e verify-sweeps`.

File: wandb/sweeps/__init__.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+from .run import next_run, next_runs, stop_runs, SweepRun, RunState  # noqa
+from .config import SweepConfig, schema_violations_from_proposed_config  # noqa
+
+__version__ = "0.0.5"
```
Comment (dannygoldstein) [inline wandb/sweeps/__init__.py:4]: this stuff is all just vendored

File: MANIFEST.in [hunk 6]
```diff
@@ -2,9 +2,10 @@ include wandb/py.typed
 include package_readme.md
 include requirements.txt
 include wandb/bin/apple_gpu_stats
-include wandb/sweeps_engine/requirements.txt
-include wandb/sweeps_engine/requirements.dev.txt
-include wandb/sweeps_engine/config/schema.json
+include wandb/sweeps/config/schema.json
+include requirements.sweeps.txt
+include .sweeps_ref
+include tools/verify_sweeps.sh
```
[the remark this answers is not in the document]
Comment (dannygoldstein) [reply MANIFEST.in:8]: This is needed for the tox build and installdeps to work correctly

[the remark this answers is not in the document]
Comment (dannygoldstein) [reply MANIFEST.in:8]: For the `verify-sweeps` environment

[the remark this answers is not in the document]
Comment (dannygoldstein) [reply MANIFEST.in]: Nevermind only `requirements.sweeps.txt` is needed because it is used by `setup.py`, the others are not needed

**Pull request merged**